src/compiler/checker.ts

```diff
@@ -8318,7 +8319,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                     // If the symbol is not found, but was not found in the original scope either we probably have an error, don't reuse the node
                     (symAtLocation === undefined && sym !== undefined) ||
                     // If the symbol is found both in declaration scope and in current scope then it shoudl point to the same reference
-                    (symAtLocation && sym && !getSymbolIfSameReference(symAtLocation, sym))
+                    (symAtLocation && sym && !getSymbolIfSameReference(symAtLocation?.exportSymbol ?? symAtLocation, sym))

```

Comment:
`getSymbolIfSameReference` should already be doing the whole "get export symbol if present" thing, I believe.

Reply:
It does merge and resolution of aliases as far as I can tell. I did not see anything about exports, and in my tests if one symbol resolved to the export and the other to the local then it returns false.

Reply:
I checked again `getSymbolIfSameReference` does not take export symbols into account.